projects/max14906/src/platform/stm32/platform_src.mk

```diff
@@ -10,6 +12,8 @@ INCS += $(PLATFORM_DRIVERS)/stm32_delay.h     \
 SRCS += $(PLATFORM_DRIVERS)/stm32_delay.c     \
 	$(PLATFORM_DRIVERS)/stm32_gpio.c      \
 	$(PLATFORM_DRIVERS)/stm32_spi.c       \
+	$(PLATFORM_DRIVERS)/stm32_pwm.c       \
```

Comment:
I think the whole idea for using HAL_TIM_MODULE_ENABLED is to not have to  include stm32_pwm.c, unless the ioc contains a timer module and it actually uses PWM for CS. this is not valid for any of the projects is No-OS. so please remove all stm32_pwm files, and no_os_pwm files from projects (added in this commit).
the same remarks applies for dma. currently no project is using spi with DMA, so why would we include those files? stm32_spi should compile without stm32_dma, if HAL_DMA_MODULE_ENABLED is not defined.

of course, you can add a project which showcases how to use SPI with DMA transfers and PWM CS. this will be use as an example (should contain DMA init and PWM init structures and the ioc should contain the modules as well).